packages/context/src/resolver.ts

```diff
@@ -136,6 +136,77 @@ function resolve<T>(
   return resolved;
 }
 
+/**
+ * Resolve entries of an object into a new object with the same keys or a
+ * promise of the new object. Please note some of the entries can be resolved
+ * synchronously while others need to be resolved asynchronously.
+ * @param map The original object containing the source entries
+ * @param resolver A function resolves an entry to a value or promise
+ */
+export function resolveMap<T, V>(
+  map: {[name: string]: T},
+  resolver: (v: T, map: {[name: string]: V}, p: string) => ValueOrPromise<V>,
+) {
```

Comment:
Could you please add an explicit type signature of the returned value? I know TypeScript is able to infer the type for us, but I think the readers of this code will appreciate the explicit type.